### ISSUE TITLE
Use of escape.Markdown for #text elements

### ISSUE THREAD

**Issue by chamilad** [JohannesKaufmann/html-to-markdown#7]
Hello,

I'm using your library for a markdown generation tool for static site generators. The Rule interface is just perfect!

The use of escape for #text elements mostly seem like a problem for me as I read through the code. Would you be able to explain why this was used in the first place? I couldn't understand why certain characters needed to be escaped in the first place. 

Thanks!

**Comment by JohannesKaufmann** [comment]
@chamilad great that you like the library!

If the following snippet gets run through the library `<p>**Not Strong**</p>` it might produce `**Not Strong**` which would not be what we are expecting. These side-effects happen with quite a few characters ("*" for bold, "_" for italic, "-" for list items, four space characters accidentally creates a code block, ...).

----

> When a header `(eg. <h3>)` contains any new lines in its body, it will split the header contents
> over multiple lines, breaking the header in Markdown (because in Markdown, a header just
> starts with #'s and anything on the next line is not part of the header). Since in HTML
> and Markdown all white space is treated the same, I chose to replace line endings with spaces.
> -> https://github.com/lunny/html2md/pull/6

With escaping, [this input](https://github.com/JohannesKaufmann/html-to-markdown/blob/810e824203714db440c4e247087cd3626f628a1f/from_test.go#L357) will generate [this output](https://github.com/JohannesKaufmann/html-to-markdown/blob/2c0dde1232402b68c74ae66fd15d6bb43d13b6fa/testdata/TestFromString/strip_newlines_from_header.golden) which is not perfect but close to the original.

---

@chamilad if you send me some snippets that behave unexpectedly, I'm happy to add some test cases and fix that. 

_As a Background Information: This library was designed to pipe whole websites through it, meaning it is supposed to handle some weird edge cases._

**Comment by estyrke** [comment]
Hi there! First, thanks for a great library! Second, I have an example that behaves unexpectedly:

The document I'm converting contains maths equations such as `<span class="tex2jax_process">$L’ = (1+n \cdot C) \cdot
    L$</span>`. Amazingly, this _almost_ works out of the box since the $<math>$ syntax is apparently used in some Markdown flavors as well. However, I get `$L’ = (1+n \\cdot C) \\cdot L$`, i.e. the backslashes before cdot are escaped. I would need them "raw": `$L’ = (1+n \cdot C) \cdot L$`.

If this is a corner case that breaks something else, then I'm happy to just write my own rule to override the default one, just thought I'd mention this.

**Comment by JohannesKaufmann** [comment]
@estyrke Yeah, you are right that is a bug. Unfortunately, it's not that easy to fix.

I have thought about a new approach that might make escaping more reliable (also resolving #19), but that requires a substantial refactor. And I don't have time for that at the moment 🤷‍♂️

---

For now, you can create a custom rule for _"span"_ and register it using `AddRules`. 

Then check whether the element has the classname _“tex2jax_process”_ using `selec.HasClass`. 

If it has return `selec.Text()` instead of `content`. That gets you the original text that is not escaped. 

If it does not have the classname, return `nil` which is then going to run the default rule.

Let me know if you have any problems...
